# Physics 21, Spring 2020
## Assignment 3: LIGO Inspirals

In [7]:
from astropy import constants as c
from matplotlib import pyplot as pl

# Question 1

In [45]:
c.M_sun

<<class 'astropy.constants.iau2015.IAU2015'> name='Solar mass' value=1.988409870698051e+30 uncertainty=4.468805426856864e+25 unit='kg' reference='IAU 2015 Resolution B 3 + CODATA 2018'>

In [42]:
c.m_n

<<class 'astropy.constants.codata2018.CODATA2018'> name='Neutron mass' value=1.67492749804e-27 uncertainty=9.5e-37 unit='kg' reference='CODATA 2018'>

In [43]:
c.m_p

<<class 'astropy.constants.codata2018.CODATA2018'> name='Proton mass' value=1.67262192369e-27 uncertainty=5.1e-37 unit='kg' reference='CODATA 2018'>

In [44]:
1.6 * c.M_sun / ((c.m_p + c.m_p) / 2)

<Quantity 1.90207706e+57>

From the assignment, we know that the mass of a neutron star is about $1-2.2$ $M_\odot$ ~ 1.6 $M_\odot$. As mass of neutron stars is almost entirely just due to protons and neutrons (which are approximately the same mass), we can write the number of nucleons in the neutron star to be $$\frac{1.6 M_\odot}{(m_n + m_p) / 2} \approx 2 \times 10^{57}$$

# Question 2

In [22]:
c.R_sun.to('km')

<Quantity 695700. km>

In [35]:
(2 * c.G * 1.4 * c.M_sun / c.c ** 2).to('km')

<Quantity 4.13455011 km>

Radius of the Sun = $6.957\times 10^5$ km. 

Radius of neutron stars ~ 10 km.

Radius (Schwarzschild) of black holes (~ 1.4 $M_\odot$) ~ 4.13 km.

# Question 3

In [46]:
c.M_sun

<<class 'astropy.constants.iau2015.IAU2015'> name='Solar mass' value=1.988409870698051e+30 uncertainty=4.468805426856864e+25 unit='kg' reference='IAU 2015 Resolution B 3 + CODATA 2018'>

In [40]:
c.G

<<class 'astropy.constants.codata2018.CODATA2018'> name='Gravitational constant' value=6.6743e-11 uncertainty=1.5e-15 unit='m3 / (kg s2)' reference='CODATA 2018'>

In [41]:
c.c

<<class 'astropy.constants.codata2018.CODATA2018'> name='Speed of light in vacuum' value=299792458.0 uncertainty=0.0 unit='m / s' reference='CODATA 2018'>

In [32]:
(c.M_sun * c.G / c.c ** 2).to("km")

<Quantity 1.47662504 km>

In [34]:
(c.M_sun * c.G / c.c ** 3).to("s")

<Quantity 4.92549095e-06 s>

Mass of Sun (in standard mass units) = $2 \times 10^{30}$ kg 

We use the conversion factor $G/c^2$. Mass of Sun (in standard length units) = $1.5$ km

We use the conversion factor $G/c^3$. Mass of Sun (in standard time units) = $5 \times 10^{-6}$ s 

# Question 4

Frequency of gravitational wave emission is twice the orbital frequency, that is $f_{GW} = 2 f_o$. As time period $T = 1/f$, we get that 0.1 s = $T_o = 1/f_o = 2 / f_{GW}$. Finally, this gives us frequency of GWs emitted is 20 Hz.

# Question 5